python/packages/kagent-core/src/kagent/core/tracing/_utils.py

```diff
@@ -27,16 +27,27 @@ def configure(fastapi_app: FastAPI | None = None):
     # Configure tracing if enabled
     if tracing_enabled:
         logging.info("Enabling tracing")
-        tracer_provider = TracerProvider(resource=resource)
         # Check new env var first, fall back to old one for backward compatibility
         trace_endpoint = os.getenv("OTEL_TRACING_EXPORTER_OTLP_ENDPOINT") or os.getenv("OTEL_EXPORTER_OTLP_ENDPOINT")
         logging.info("Trace endpoint: %s", trace_endpoint or "<default>")
         if trace_endpoint:
             processor = BatchSpanProcessor(OTLPSpanExporter(endpoint=trace_endpoint))
         else:
             processor = BatchSpanProcessor(OTLPSpanExporter())
-        tracer_provider.add_span_processor(processor)
-        trace.set_tracer_provider(tracer_provider)
+
+        # Check if a TracerProvider already exists (e.g., set by CrewAI)
+        current_provider = trace.get_tracer_provider()
+        if isinstance(current_provider, TracerProvider):
+            # TracerProvider already exists, just add our processor to it
```

Comment:
I think we also need to add our labels here, e.g. `Resource({"service.name": "kagent"})`